lib/compat/wordpress-6.2/get-global-styles-and-settings.php

```diff
@@ -12,11 +12,16 @@
 	 * The result would be cached via the WP_Object_Cache.
 	 * It can be cleared by calling wp_theme_has_theme_json_clean_cache().
 	 *
+	 * @param string $stylesheet Directory name for the theme. Optional. Defaults to current theme.
+	 *
 	 * @return boolean
 	 */
-	function wp_theme_has_theme_json() {
-		$cache_group       = 'theme_json';
-		$cache_key         = 'wp_theme_has_theme_json';
+	function wp_theme_has_theme_json( $stylesheet = '' ) {
+		$cache_group = 'theme_json';
+		if ( empty( $stylesheet ) ) {
+			$stylesheet = get_stylesheet();
+		}
+		$cache_key         = sprintf( 'wp_theme_has_theme_json_%s', $stylesheet );
```

Comment:
I see two issues with caching by theme:

- Code complexity: what are the benefits of having cache keys per theme when we only need one at a time?
- I don't see a way to clean the cache key for the previous theme upon a `start_previewing_event`. It gives the stylesheet of the theme being previewed, but at that point we need to clear the cache key of the _previous_ theme, not the one being previewed. How can we do this?

Reply:
Complexity is not really a good argument not to do something. Just becuase it is hard, does not mean it not worthwhile. I have done the hard work to make this work, let's go ahead here. 

Good point about `start_previewing_event`. This cache clearing can be removed, as now the cache key will be different once you start previewing, which is pretty neat. I am not against removing this all together, but I kept, just in case.

Reply:
> Complexity is not really a good argument not to do something. Just becuase it is hard, does not mean it not worthwhile. I have done the hard work to make this work, let's go ahead here.

My point is not about doing something that is hard. It is about not doing something complex when it can be simple.

This approach is more complex in terms of code and more demanding in terms of memory (we store cache keys we don't need instead of one). The direction I shared a while ago and summarized [here](https://github.com/WordPress/gutenberg/pull/45955/files#r1029306186) again is simpler and brings bigger performance gains. Would you be willing to give it a try and reevaluate when is it shipped?

Reply:
I don't understand this feedback at all. For most requests, it will only store one cache key, like it does currently. There difference here, is this structure all for multiple themes theme json to be required. This makes cache invalidation easier, not harder. 

Where is the there a performance lose here?